src/object.c

```diff
@@ -1173,15 +1173,31 @@ struct redisMemOverhead *getMemoryOverheadData(void) {
     mem_total += server.initial_memory_usage;
 
     mem = 0;
-    if (server.repl_backlog)
-        mem += zmalloc_size(server.repl_backlog);
+    if (server.repl_backlog && server.repl_backlog->ref_repl_buf_node) {
+        replBufBlock *head = listNodeValue(server.repl_backlog->ref_repl_buf_node);
+        replBufBlock *last = listNodeValue(listLast(server.repl_buffer_blocks));
+        // TODO COMMENTS
+        /*
+        maybe split this into several mem+= lines each with a comment.
+i understand the last two compute the rax,
+the second one the linked list (shared with replicas, right?).
+and the first one computes the actual bytes.
+and since the histlen counts the used portion, we need to explicitly add the unused portion..
+        */
+        mem += server.repl_backlog_histlen + (last->size - last->used);
+
+        mem += (last->id - head->id + 1) * (sizeof(replBufBlock)+sizeof(listNode));
+
+        mem += server.repl_backlog->blocks_index->numnodes * sizeof(raxNode) +
+               raxSize(server.repl_backlog->blocks_index) * sizeof(void*);
+    }
     mh->repl_backlog = mem;
     mem_total += mem;
 
+    mh->clients_slaves = getSlavesOutputBufferMemoryUsage();
```

Comment:
i had a discussion about this with @yoav-steinberg and although we don't agree, i'd list the options here.

background:
previously we had these in INFO MEMORY: `mem_clients_slaves` and `mem_replication_backlog`, people had to sum them if they wanted, or alternatively looking at each (specifically the slaves one), would immediately indicate where the majority of the memory is used.

now we have several options:
1. keep each of these two represent how much memory used for that purpose, so summing them would mean part of the memory was accounted twice, and that's what the new `mem_total_replication_buffers` field is for.
2. delete these old fields since they're coupled with an old internal design. users can easily conclude that if the new `mem_total_replication_buffers` is bigger than the `repl_backlog_size` config, then that's an indication that replicas are causing trouble.
3. we can aim for 2, but meanwhile keep doing 1 for some backwards compatibility.
4. do what i'm gonna describe below:

my suggestion:
```
if (server.repl_buffer_mem > server.repl_backlog_histlen) {
    mem_clients_slaves = server.repl_buffer_mem - server.repl_backlog_histlen;
    mem_replication_backlog =  server.repl_backlog_histlen;
} else {
    mem_clients_slaves = 0;
    mem_replication_backlog =  server.repl_backlog_histlen;
}
```
- so in case the replicas are not lagging, they'll show 0, and even if they do consume something, then we can argue that the new mechanism is utilizing the backlog so it doesn't cause any excess memory usage.
- in case the replicas are lagging, the backlog will show the maximum, and the replicas will show the excess.
- this is backwards compatible with what we had till now (people can sum these fields)
- a quick look at these two metrics, and it's easy to realize where the memory is wasted and who's to blame

Reply:
we may need a tie breaker (or a 5th option).
@yossigo please weigh in (or call me for quick summary).

Reply:
My take on this is it really depends if the user is trying to figure out how memory is utilized, or figure out how replication is doing. @oranagra pointed out that it's expected that `INFO memory` would focus on memory usage because more detailed replica information can be accessed by `INFO replication` or `CLIENT LIST`. If we follow this reasoning then [4] makes sense.

The main arguments I see against option [4] are:
* It's usually best to expose raw metrics and not computed ones
* Changing the meaning of an existing metric is bad practice.

Perhaps a 5th option is [4] but using a different field name to indicate we're exposing the extra memory that does not overlap the backlog.